src/components/Dropdown/Dropdown.Props.ts

```diff
@@ -58,4 +58,9 @@ export interface IDropdownOption {
    * Whether this option is currently selected.
    */
   isSelected?: boolean;
+
+  /**
+   * Optional custom renderer for the dropdown item
+   */
+  onRender?: (item: IDropdownOption) => JSX.Element;

```

Comment:
i think this would make it rather clunky to use. Wouldn't it be a little cleaner to just have an onRenderItem on the dropdown itself? Then you could abstract this, just give it data items and let the rendering be managed in one place.